data-reporting-service/common-util/src/main/java/gov/cdc/etldatapipeline/commonutil/json/CustomJsonGeneratorImpl.java

```diff
@@ -0,0 +1,110 @@
+                } else {
+                    dataField.setField(PropertyNamingStrategies.SnakeCaseStrategy.INSTANCE.translate(field.getName()));
+                }
+                dataField.setOptional(
+                        obj.getRequiredFields() == null || !obj.getRequiredFields().contains(field.getName()));
+                dataField.setType(getType(field.getType().getSimpleName().toLowerCase()));
+                dataFields.add(dataField);
+            }
+        } catch (Exception e) {
+            throw new RuntimeException("Error building schema record: ", e);
+        }
+        return new DataEnvelope(new DataSchema("struct", dataFields), obj);
+    }
+
+    public String generateStringJson(Object model) {
+        try {
+            ObjectMapper objectMapper = new ObjectMapper().registerModule(new JavaTimeModule());
+            ObjectNode root = objectMapper.createObjectNode();
+            ObjectNode schemaNode = root.putObject("schema");
+            schemaNode.put("type", "struct");
+            schemaNode.set("fields", generateFieldsArray(model));
+            ObjectNode payloadNode = root.putObject("payload");
+            payloadNode = generatePayloadNode(payloadNode, model);
+            return objectMapper.writeValueAsString(root);
+        } catch (Exception e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    private static ArrayNode generateFieldsArray(Object model) {
+        ObjectMapper objectMapper = new ObjectMapper().registerModule(new JavaTimeModule());
+        ArrayNode fieldsArray = objectMapper.createArrayNode();
+
+        try {
+            Class<?> modelClass = model.getClass();
+            for (Field field : modelClass.getDeclaredFields()) {
+                ObjectNode fieldNode = objectMapper.createObjectNode();
+
+                String fieldName = field.getName();
+
+                fieldName = CaseFormat.LOWER_CAMEL.to(CaseFormat.LOWER_UNDERSCORE, fieldName);
+
+                fieldNode.put("type", getType(field.getType().getSimpleName().toLowerCase()));
+                fieldNode.put("optional", !fieldName.equals("public_health_case_uid"));
+                fieldNode.put("field", fieldName);
+                fieldsArray.add(fieldNode);
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        return fieldsArray;
+    }
+
+    private static ObjectNode generatePayloadNode(ObjectNode payloadNode, Object model) {
+        ObjectMapper objectMapper = new ObjectMapper().registerModule(new JavaTimeModule());
+
+        try {
+            Class<?> modelClass = model.getClass();
+            for (java.lang.reflect.Field field : modelClass.getDeclaredFields()) {
+                field.setAccessible(true);
+                String fieldName = field.getName();
+
+                fieldName = CaseFormat.LOWER_CAMEL.to(CaseFormat.LOWER_UNDERSCORE, fieldName);
```

Comment:
Does this require the fieldName to be in camelCase or any such restriction.

Reply:
Some unit tests to validate this will be good.

Reply:
I'll create a separate story for unit tests